metadoc-js/src/main/scala/metadoc/ScalaReferenceProvider.scala

```diff
@@ -1,31 +1,51 @@
 package metadoc
 
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 import scala.scalajs.js
 import scala.scalajs.js.annotation._
 import scala.meta.Attributes
 import metadoc.schema.Index
 import monaco.{CancellationToken, Position}
 import monaco.editor.IReadOnlyModel
 import monaco.languages.{Location, ReferenceContext, ReferenceProvider}
+import metadoc.{schema => d}
+import monaco.editor.IModel
 
 @ScalaJSDefined
-class ScalaReferenceProvider(attrs: Attributes, index: Index)
-    extends ReferenceProvider {
+class ScalaReferenceProvider(index: Index) extends ReferenceProvider {
   override def provideReferences(
       model: IReadOnlyModel,
       position: Position,
       context: ReferenceContext,
       token: CancellationToken
   ) = {
     val offset = model.getOffsetAt(position)
-    val positions = IndexLookup.findReferences(
-      offset,
-      context.includeDeclaration,
-      attrs,
-      index,
-      model.uri.path
-    )
-    val locations = positions.map(resolveLocation(model))
-    js.Array[Location](locations: _*)
-  }
+    for {
+      attrs <- MetadocAttributeService.fetchAttributes(model.uri.path)
+      id = IndexLookup.findSymbol(offset, attrs, index).map(_.symbol)
+      // Monad transformers might come in handy here.
```

Comment:
Totally agree! I would love to move this logic to `core` and test it properly on the JVM.